pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/json/MutableJsonIndexImpl.java

```diff
@@ -209,41 +210,11 @@ private RoaringBitmap getMatchingFlattenedDocIds(Predicate predicate) {
     } else {
       key = JsonUtils.KEY_SEPARATOR + key;
     }
-
```

Comment:
Move the processing of array index to the static function processArrayIndex for code reusing